packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -6,32 +6,97 @@ import { some, groupBy } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { CheckboxControl, Modal, Button } from '@wordpress/components';
+import {
+	CheckboxControl,
+	Button,
+	PanelBody,
+	PanelRow,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { useSelect, useDispatch } from '@wordpress/data';
-import { useState } from '@wordpress/element';
+import { useState, useCallback } from '@wordpress/element';
+import { close, page, layout, grid, blockDefault } from '@wordpress/icons';
+
+const ENTITY_NAME_ICONS = {
+	site: layout,
+	page,
+	post: grid,
+	wp_template: grid,
+};
+
+function EntityRecordState( { record, checked, onChange, closePanel } ) {
+	const { name, kind, title, key } = record;
+	const parentBlockId = useSelect( ( select ) => {
+		// Get entity's blocks.
+		const { blocks = [] } = select( 'core' ).getEditedEntityRecord(
+			kind,
+			name,
+			key
+		);
+		// Get parents of the entity's first block.
+		const parents = select( 'core/block-editor' ).getBlockParents(
+			blocks[ 0 ]?.clientId
+		);
+		// Return closest parent block's clientId.
+		return parents[ parents.length - 1 ];
+	}, [] );
+
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+	const selectParentBlock = useCallback( () => selectBlock( parentBlockId ), [
+		parentBlockId,
+	] );
+
+	const selectAndDismiss = useCallback( () => {
+		selectBlock( parentBlockId );
+		closePanel();
+	}, [ parentBlockId ] );
 
-function EntityRecordState( { record, checked, onChange } ) {
 	return (
-		<CheckboxControl
-			label={ <strong>{ record.title || __( 'Untitled' ) }</strong> }
-			checked={ checked }
-			onChange={ onChange }
-		/>
+		<PanelRow>
+			<CheckboxControl
+				label={ <strong>{ title || __( 'Untitled' ) }</strong> }
+				checked={ checked }
+				onChange={ onChange }
+			/>
+			{ parentBlockId ? (
+				<>
+					<Button
+						onClick={ selectParentBlock }
+						className="entities-saved-states__find-entity"
+					>
+						{ __( 'Select Entity' ) }
```

Comment:
This is super interesting. It seems like a fun feature to experiment with. Perhaps the copy should just be "Select" for now.

Reply:
Thanks!  It was inspired by your mock-up 😄 - I just have it as a separate button temporarily since we are still using the checkbox controls for now.

> Perhaps the copy should just be "Select" for now.

I am agreeable to that, I could see a general using thinking "what the heck is an entity?".  Pushing that change now.